src/transport/msg_providers/onchain_transport.py

```diff
@@ -0,0 +1,255 @@
+import abc
+import logging
+from enum import StrEnum
+from typing import List, Optional
+
+from eth_typing import ChecksumAddress
+from schema import Schema
+from transport.msg_providers.common import BaseMessageProvider
+from transport.msg_providers.rabbit import MessageType
+from transport.msg_types.deposit import DepositMessage
+from transport.msg_types.pause import PauseMessage
+from transport.msg_types.ping import PingMessage
+from transport.msg_types.unvet import UnvetMessage
+from utils.bytes import bytes_to_hex_string
+from web3 import Web3
+from web3._utils.events import get_event_data
+from web3.exceptions import BlockNotFound
+from web3.types import EventData, FilterParams, LogReceipt
+
+logger = logging.getLogger(__name__)
+
+# event Message(
+#     bytes32 indexed eventId,
+#     address indexed sender,
+#     bytes data
+# ) anonymous;
+MESSAGE_EVENT_ABI = {
```

Comment:
```suggestion
EVENT_ABI = {
```

Reply:
Let's split two definitions:
EVENT_ABI - general abi applied for all events in contract
MESSAGE_ABI - unique for each message type abi

Reply:
Done